Src/AutoFixtureUnitTest/MailAddressGeneratorTests.cs

```diff
@@ -62,5 +87,32 @@ public void CreateWithMailAddressRequestReturnsCorrectResult()
             Assert.True(Regex.IsMatch(result.Host, @"example\.(com|org|net)"));
             // Teardown
         }
+
+        [Fact]
+        public void CreateWithMailAddressRequestReturnsNoSpecimenWhenContextReturnsNullLocalPart()
+        {
+            // Fixture setup
+            var request = typeof(MailAddress);
+            var expectedLocalPart = new EmailAddressLocalPart(Guid.NewGuid().ToString());
+            var context = new DelegatingSpecimenContext()
+            {
+                OnResolve = r =>
+                {
+                    if (typeof(EmailAddressLocalPart).Equals(r))

```

Comment:
How do we feel about [Conditional Test Logic](http://xunitpatterns.com/Conditional%20Test%20Logic.html)?